src/ip/Intercept.cc

```diff
@@ -428,6 +417,41 @@ Ip::Intercept::Lookup(const Comm::ConnectionPointer &newConn, const Comm::Connec
     return false;
 }
 
+
+bool
+Ip::Intercept::LookupTproxy(const Comm::ConnectionPointer &newConn)
+{
+    /* --enable-linux-netfilter    */
+    /* --enable-ipfw-transparent   */
+    /* --enable-ipf-transparent    */
+    /* --enable-pf-transparent     */
+#if IPF_TRANSPARENT || LINUX_NETFILTER || IPFW_TRANSPARENT || PF_TRANSPARENT
+
+    debugs(89, 5, HERE << "address BEGIN: me/client= " << newConn->local << ", destination/me= " << newConn->remote);
+
+    /* NP: try TPROXY first, its much quieter than NAT when non-matching */
+    if (transparentActive_ && TproxyTransparent(newConn))
+        return true;
+
+#else /* none of the transparent options configured */
+    debugs(89, DBG_IMPORTANT, "WARNING: transparent proxying not supported");
```

Comment:
AFAICT, this whole preprocessor logic should be removed from this method, and the warning moved to TproxyTransparent(). It is OK to always call level-5 debugs() above, regardless of the `#if` guards.